crates/viewer/re_space_view_dataframe/src/space_view_class.rs

```diff
@@ -77,6 +98,20 @@ for all entities, it is preferable to override the view-level visible time range
         re_viewer_context::SpaceViewClassLayoutPriority::Low
     }
 
+    fn default_query_range(&self, state: &dyn SpaceViewState) -> QueryRange {
+        // TODO(#6918): passing the mode via view state is a hacky work-around, until we're able to
+        // pass more context to this function.
+        let mode = state
+            .downcast_ref::<DataframeViewState>()
+            .map(|state| state.mode)
+            .unwrap_or_default();
```

Comment:
Every `downcast()` should log a warning on the failure path or this will silently fail for who knows how long if someone unknowingly breaks the code upstream.